debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/BufferedMemory.java

```diff
@@ -0,0 +1,225 @@
+import com.ibm.j9ddr.corereaders.memory.IMemorySource;
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.DataUnavailableException;
+import com.ibm.j9ddr.corereaders.osthread.IOSThread;
+import com.ibm.j9ddr.corereaders.ICore;
+import com.ibm.j9ddr.corereaders.memory.SymbolUtil;
+
+/**
+ * Object representing a single live
+ * process model where a section of
+ * memory in a the process can be represented
+ * in a byte buffer as its source
+ * @see com.ibm.j9ddr.tools.ddrinteractive.BufferedMemorySource
+ *
+ * Can be used for inspection of the
+ * scc, without the need of a core file
+ * For example in the following, it is
+ * used to inspect romMethods:
+ *
+ * <code>
+ * BufferedMemory memory = new BufferedMemory(ByteOrder.nativeOrder());
+ * IVMData aVMData = VMDataFactory.getVMData((IProcess)memory);
+ * //this address denotes the beginning of range of interest, not shown here
+ * memory.addMemorySource(new BufferedMemorySource(sourceStartAddress, size));
+ * aVMData.bootstrap("com.ibm.j9ddr.vm29.SomeDebugHandler");
+ *
+ * public class SomeDebugHandler {
+ *   //can fetch this address from a ROMClass Cookie
+ *   J9ROMClassPointer pointer = J9ROMClassPointer.cast(romStartAddress);
+ *   J9ROMMethodPointer romMethod = pointer.romMethods();
+ *   long dumpFlags = (ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN) ? 1 : 0;
+ *   J9BCUtil.j9bcutil_dumpRomMethod(System.out, romMethod, pointer, dumpFlags, J9BCUtil.BCUtil_DumpAnnotations);
+ * }
+ * </code>
+ *
+ * Behaviour of the model is not defined
+ * for the case where the underlying source's
+ * ByteBuffer is modified (address or capacity)
+ * during use.
+ *
+ * @author knewbury01
+ *
+ */
+
+public class BufferedMemory extends AbstractMemory implements IProcess, IAddressSpace
+{
+	public BufferedMemory(ByteOrder byteOrder) {
+		super(byteOrder);
+	}
+
+	@Override
+	public Platform getPlatform() {
+		String platform = System.getProperty("os.name").toLowerCase();
+		if (platform.contains("aix")) {
+			return Platform.AIX;
+		}
+		else if (platform.contains("windows")) {
+			return Platform.WINDOWS;
+		}
+		else if	(platform.contains("z/os")) {
+			return Platform.ZOS;
+		}
+		else if	(platform.contains("linux")) {
+			return Platform.LINUX;
+		}
+		else if (platform.contains("mac")) {
+			return Platform.OSX;
+        }
+		else {
+			//do not expect to reach here
+			return null;
+		}
+	}
+
+	/**
+	 *
+	 * @return Address space this process uses.
+	 */
+	@Override
+	public IAddressSpace getAddressSpace() {
+		return this;
+	}
+
+	@Override
+	public long getPointerAt(long address) throws MemoryFault {
+		if (bytesPerPointer() == 8) {
+			return getLongAt(address);
+		} else {
+			return (0xFFFFFFFFL & getIntAt(address));
+		}
+	}
+
+	/**
+	 *
+	 * @return Number of bytes in a pointer
+	 */
+	@Override
+	public int bytesPerPointer() {
+		return Integer.parseInt(System.getProperty("sun.arch.data.model"));
```

Comment:
This isn't a practical implementation; this method will be called frequently fetching and parsing the property every time will make things slower than necessary. Please move the calculation into the constructor and save the result in a final field.